pkg/kube/helpers.go

```diff
@@ -215,6 +215,24 @@ func mutate(f MutateFn, key client.ObjectKey, obj client.Object) error {
 // TODO this code is from https://github.com/kubernetes/kubernetes/blob/master/pkg/api/v1/pod/util.go
 // We need to determine if this functionality is available via the client-go
 
+// IsPodAvailable returns true if a pod is available; false otherwise.
+// Precondition for an available pod is that it must be ready. On top
+// of that, there are two cases when a pod can be considered available:
+// 1. minReadySeconds == 0, or
+// 2. LastTransitionTime (is set) + minReadySeconds < current time
+func IsPodAvailable(pod *corev1.Pod, minReadySeconds int32, now metav1.Time) bool {
```

Comment:
@chrisseto this seems to still fail.  I am getting db connection issues, and the only thing I can think of is that the pod is not up and running correctly.  Thoughts?

Reply:
Seems like you might be grabbing the wrong pods?